test/utils.ts

```diff
@@ -31,12 +31,12 @@ export const toBN = (num: string | number | BigNumber) => {
 };
 
 export async function seedWallet(
-  walletToFund: SignerWithAddress,
+  walletToFund: Signer,
   tokens: Contract[],
   weth: Contract | undefined,
   amountToSeedWith: number | BigNumber
 ) {
-  for (const token of tokens) await token.mint(walletToFund.address, amountToSeedWith);
+  for (const token of tokens) await token.mint(await walletToFund.getAddress(), amountToSeedWith);
```

Comment:
why this change?

Reply:
SignerWithAddress is more restrictive than Signer. It's pretty easy to get the address from the Signer and the Signer type is much easier to extract from the ethers package, so I think we should prefer using it over the more specific and cumbersome SignerWithAddress when we're building generic library functions.